dev-utils/src/commands/genesis_command.rs

```diff
@@ -18,84 +25,268 @@ use crate::commands::genesis_dev_config::GenesisDevConfig;
+        (
+            ErrorKind::InvalidInput,
+            "Something went wrong reading the Tendermint genesis file",
+        )
+    })?;
+    let mut tendermint_genesis: serde_json::Value =
+        serde_json::from_str(&tendermint_genesis_config).chain(|| {
+            (
+                ErrorKind::DeserializationError,
+                "failed to parse Tendermint genesis file",
+            )
+        })?;
+
+    let genesis_dev_config_string = fs::read_to_string(genesis_dev_config_path).chain(|| {
+        (
+            ErrorKind::InvalidInput,
+            "Something went wrong reading the genesis dev config file",
+        )
+    })?;
+    let genesis_dev_config: GenesisDevConfig = serde_json::from_str(&genesis_dev_config_string)
+        .chain(|| {
+            (
+                ErrorKind::DeserializationError,
+                "failed to parse genesis dev config",
+            )
+        })?;
+
+    let (app_hash, app_state, validators) = generate_genesis(&genesis_dev_config)?;
+
+    let app_hash = serde_json::to_value(app_hash).chain(|| {
+        (
+            ErrorKind::SerializationError,
+            "failed to convert generated app hash into json value",
+        )
+    })?;
+    let app_state = serde_json::to_value(app_state).chain(|| {
+        (
+            ErrorKind::SerializationError,
+            "failed to convert generated app state into json value",
+        )
+    })?;
+    let validators = serde_json::to_value(validators).chain(|| {
+        (
+            ErrorKind::SerializationError,
+            "failed to convert generated validators into json value",
+        )
+    })?;
+    tendermint_genesis["app_hash"] = app_hash;
+    tendermint_genesis["app_state"] = app_state;
+    tendermint_genesis["validators"] = validators;
+
+    let tendermint_genesis_string =
+        serde_json::to_string_pretty(&tendermint_genesis).chain(|| {
             (
                 ErrorKind::InvalidInput,
-                "Something went wrong reading the file",
+                "Invalid generated Tendermint genesis",
             )
         })?;
-        let genesis_dev: GenesisDevConfig =
-            serde_json::from_str(&genesis_dev_config).expect("failed to parse genesis dev config");
 
-        GenesisCommand::do_generate(&genesis_dev)
+    if inline {
+        backup_tendermint_genesis(&tendermint_genesis_path)?;
+        write_tendermint_genesis(&tendermint_genesis_path, &tendermint_genesis_string)?;
+    } else {
+        println!("{}", tendermint_genesis_string);
     }
+
+    Ok(())
+}
+
+fn find_default_tendermint_path() -> Option<PathBuf> {
+    find_tendermint_path_from_env("TMHOME")
+        .or_else(|| find_tendermint_path_from_env("HOME"))
+}
+
+fn find_tendermint_path_from_env(env_name: &str) -> Option<PathBuf> {
+    if let Ok(home) = env::var(env_name) {
+        let path_buf = PathBuf::from(format!("{}/.tendermint/config/genesis.json", home));
```

Comment:
Default `TMHOME` is `$HOME/.tendermint`. You don’t have to add `.tendermint` to path when `TMHOME` is present in environment.

Reply:
Good catch, I will change it.

Reply:
Done